### PR TITLE
Move loop into map. Second version

### PR DESCRIPTION
**Pull request by PanicSheep**

### CODE REVIEW

**Review by tbennun — CHANGES_REQUESTED** [review]

File: dace/transformation/interstate/move_loop_into_map.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        
+        for node in body.nodes():
+            if isinstance(node, nodes.MapEntry):
+                map_entry = node
+            if isinstance(node, nodes.MapExit):
+                map_exit = node
+
+        # save old in and out memlets { name : ranges }
+        old_in_memlets = { memlet.data.data : memlet.data.subset.ranges for memlet in body.out_edges(map_entry) }
+        old_out_memlets = { memlet.data.data : memlet.data.subset.ranges for memlet in body.in_edges(map_exit) }
```
[the remark this answers is not in the document]
Comment (tbennun) [reply dace/transformation/interstate/move_loop_into_map.py]: (1) was addressed, (2) wasn't.

[the remark this answers is not in the document]
Comment (alexnick83) [reply dace/transformation/interstate/move_loop_into_map.py]: This code is obsolete and was removed.

File: dace/transformation/interstate/move_loop_into_map.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        after: sd.SDFGState = sdfg.node(candidate[DetectLoop._exit_state])
+
+        # Obtain iteration variable, range, and stride
+        itervar, (start, end, step), _ = find_for_loop(sdfg, guard, body)
+
+        if step not in [-1, 1]:
+            return False
+
+        # Check if body contains exactly one map
+        is_map = [isinstance(node, nodes.MapEntry) for node in body.nodes()]
```
[the remark this answers is not in the document]
Comment (alexnick83) [reply dace/transformation/interstate/move_loop_into_map.py]: These conditions were amended.

**Review by alexnick83 — APPROVED** [review]
LGTM

**Pull request merged**